PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/filesystem/StorageChooser.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.microsoft.azure.hdinsight.spark.ui.filesystem;
+
+import com.intellij.openapi.fileChooser.FileChooserDescriptor;
+import com.intellij.openapi.fileChooser.FileChooserDialog;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.microsoft.azure.hdinsight.common.logger.ILogger;
+import com.microsoft.azure.hdinsight.sdk.storage.adlsgen2.ADLSGen2FSOperation;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
+
+import java.awt.*;
+import java.net.URI;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+public class StorageChooser implements ILogger {
+    private AzureStorageVirtualFileSystem fileSystem;
+
+    public StorageChooser(@Nullable AzureStorageVirtualFileSystem fileSystem) {
+        this.fileSystem = fileSystem;
+    }
+
+    public List<VirtualFile> setRoots(String uploadRootPath) {
+        if (fileSystem == null) {
+            return Arrays.asList(AdlsGen2VirtualFile.Empty);
+        }
+
+        String rootPath = uploadRootPath.replace("/SparkSubmission/", "");
```

Comment:
`SparkSubmission`'s parent may not be the root

Reply:
in gen2 api ,dns name is needed others should be in param 

---
In reply to: [285430757](https://github.com/microsoft/azure-tools-for-java/pull/3124#discussion_r285430757) [](ancestors = 285430757)

Reply:
we can change the var name

---
In reply to: [285555149](https://github.com/microsoft/azure-tools-for-java/pull/3124#discussion_r285555149) [](ancestors = 285555149,285430757)